pandas/core/accessor.py

```diff
@@ -127,6 +156,11 @@ def delegate_names(delegate, accessors, typ: str, overwrite: bool = False):
     typ : {'property', 'method'}
     overwrite : bool, default False
        Overwrite the method/property in the target class if it exists.
+    accessor_mapping: Callable, default lambda x: x
+        Callable to map the delegate's function to the cls' function.
+    raise_on_missing: bool, default True
+        Raise if an accessor does not exist on delegate.
+        False skips the missing accessor.
```

Comment:
do you expect to reuse this machinery again?  if not, could the pyarrow-dt be accessor be implemented standalone to avoid making this more complicaed?

Reply:
I am hoping to use this for the pyarrow duration types as well.

I think a standalone implementation would require a further refactoring of this `dt` accessor. Looks like currently datetime and timedelta dispatch to the Index while we probably need something that dispatches to the array